6-anthos-install/README.md

```diff
@@ -0,0 +1,346 @@
+
+You can also connect to this instance by tunnelling SSH traffic through IAP.
+```console
+# Replace YOUR_PROJECT_ID with your GKE project ID.
+export PROJECT_ID=YOUR_PROJECT_ID
+gcloud compute ssh gce-bastion-us-west1-b-01 \
+  --project ${PROJECT_ID}
+```
+
+## Insall required tools
+```console
+sudo su
+yum install git
+yum install google-cloud-sdk-kpt
+yum install jq -y
+yum install kubectl
+exit
+```
+
+## Pull the repo
+```console
+cd ${HOME}
+git clone https://github.com/GoogleCloudPlatform/terraform-example-foundation-app.git
+```
+
+## Define required environment variables
+When indicated, make sure to replace the values below with the appropriate values based on the outcome of terraform.
+```console
+# replace YOUR_PROJECT_ID with the project id for the project that hosts your clusters.
+# For example: prj-bu1-d-boa-gke-ecb0
+export PROJECT_ID=YOUR_PROJECT_ID
+export PROJECT_NUM=$(gcloud projects describe ${PROJECT_ID} --format='value(projectNumber)')
+export CLUSTER_1=gke-boa-us-east1-001
+export CLUSTER_1_REGION=us-east1
+export CLUSTER_2=gke-boa-us-west1-001
+export CLUSTER_2_REGION=us-west1
+export CLUSTER_INGRESS=gke-mci-us-east1-001
+export CLUSTER_INGRESS_REGION=us-east1
+export WORKLOAD_POOL=${PROJECT_ID}.svc.id.goog
+export MESH_ID="proj-${PROJECT_NUM}"
+export ASM_VERSION=1.8
+export ISTIO_VERSION=1.8.3-asm.2
+export ASM_LABEL=asm-183-2
+export CTX_1=$PROJECT_ID_$CLUSTER_1_REGION_CLUSTER_1
+export CTX_2=$PROJECT_ID_$CLUSTER_2_REGION_CLUSTER_2
+export CTX_INGRESS=$PROJECT_ID_$CLUSTER_INGRESS_REGION_CLUSTER_INGRESS
+```
+## Generate Kubeconfig Entries
+In order to install ASM, we need to authenticate to clusters.
+
+```console
+gcloud container clusters get-credentials ${CLUSTER_1} --region ${CLUSTER_1_REGION}
+gcloud container clusters get-credentials ${CLUSTER_2} --region ${CLUSTER_2_REGION}
+gcloud container clusters get-credentials ${CLUSTER_INGRESS} --region ${CLUSTER_INGRESS_REGION}
+```
+## Install ASM
+### Downloading the script
+
+1. Download the version of the script that installs ASM 1.8.3.
+    ```console
+    curl https://storage.googleapis.com/csm-artifacts/asm/install_asm_"${ASM_VERSION}" > install_asm
+    ```
+1. Make the script executable:
+    ```console
+    chmod +x install_asm
+    ```
+1. Create a folder to host the installation files and asm packages. This folder will also include `istioctl`, sample apps, and manifests.
+    ```console
+    mkdir -p ${HOME}/asm-${ASM_VERSION} && export PATH=$PATH:$HOME/asm-${ASM_VERSION}
+    ```
+
+### Install ASM on both clusters
+The following commands run the script for a new installation of ASM on Cluster1 and Cluster2. By default, ASM uses Mesh CA. The `--enable_cluster_labels` flag allows the script to attempt to bind the service account running the script to the cluster-admin role on the cluster. `--enable_cluster_roles` flag allow the script to set the required cluster labels.
```

Comment:
--enable_cluster_labels 
This only sets labels mesh_id and asmv. Labels can be set on the cluster in terraform, happy to make the change.
`mesh_id = "proj-${PROJECT_NUMBER}"` 
could not find anything definitive on what the value for asmv should be nor does it seem necessary for the functionality

--enable_cluster_roles
the bastion service account can be given cluster admin unless you want the user running the script to get the role
`gcloud config set core/account boa-gce-bastion-x-sa@prj-bu1-x-boa-gke-xxxx.iam.gserviceaccount.com`

Reply:
>  Labels can be set on the cluster in terraform

yes we should like https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/1fe90e9e99112ad3fcbd3f87fcc652b7c38ee3bf/examples/simple_zonal_with_asm/main.tf#L57
With https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages/pull/454 asmv should not be necessary

> the bastion service account can be given cluster admin

IIRC enable_cluster_roles was for k8s clusterrolebinding and not GCP IAM. If it is for k8s, we should enable via flag for now.

Reply:
Yes, it's for k8s